File: pdu/python/hako_msgs/pdu_cdr_conv_HakoStatusMagnetHolder.py
from .pdu_pytype_HakoStatusMagnetHolder import HakoStatusMagnetHolder
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_HakoStatusMagnetHolder(writer: CdrWriter, src: HakoStatusMagnetHolder):
    writer.write_bool(src.magnet_on)
    writer.write_bool(src.contact_on)


def cdr_body_to_py_HakoStatusMagnetHolder(reader: CdrReader, dst: HakoStatusMagnetHolder):
    dst.magnet_on = reader.read_bool()
    dst.contact_on = reader.read_bool()
    return dst


def py_to_cdr_HakoStatusMagnetHolder(src: HakoStatusMagnetHolder) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_HakoStatusMagnetHolder(writer, src)
    return writer.bytes()


def cdr_to_py_HakoStatusMagnetHolder(cdr_payload) -> HakoStatusMagnetHolder:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = HakoStatusMagnetHolder()
    return cdr_body_to_py_HakoStatusMagnetHolder(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_HakoStatusMagnetHolder.js
import { HakoStatusMagnetHolder } from './pdu_jstype_HakoStatusMagnetHolder.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduHakoStatusMagnetHolderConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { HakoStatusMagnetHolder } src
     */
    static to_cdr_body(writer, src) {
        writer.write_bool(src.magnet_on);
        writer.write_bool(src.contact_on);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { HakoStatusMagnetHolder } dst
     * @returns { HakoStatusMagnetHolder }
     */
    static cdr_body_to_js(reader, dst) {
        dst.magnet_on = reader.read_bool();
        dst.contact_on = reader.read_bool();
        return dst;
    }

    /**
     * @param { HakoStatusMagnetHolder } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { HakoStatusMagnetHolder }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new HakoStatusMagnetHolder());
    }
}
